### PR TITLE
Clarify and link cloud/security goals

### PR DESCRIPTION
**Pull request by nicksnyder**
This is an attempt to more clearly document the goals of the Cloud and Security team, as well as make it clear how they relate to each other.

Some of the milestones in the security team goals are still rough, and I want to get some clarifications (questions are inline).

### CODE REVIEW

File: handbook/engineering/cloud/index.md
Comment (tsenart) [inline handbook/engineering/cloud/index.md:50]:
```suggestion
**Problem:** Customers who want to use Sourcegraph with their private code must setup and run a Sourcegraph instance on their own compute infrastructure. There are customers who want to use Sourcegraph but don't want to have to deploy and operate their own Sourcegraph instance and associated compute infrastructure. To host private code on Sourcegraph Cloud, we need to not only ensure the security of our product (which is important for on-premise deployments too), but we also have to ensure the security of our Sourcegraph Cloud infrastructure.
```

File: handbook/engineering/security/index.md
Comment (tsenart) [inline handbook/engineering/security/index.md:20]: We should not be using LogDNA — that was an experiment that @keegancsmith started. I think we should keep our logs, which may contain sensitive data, inside our core infrastructure (i.e GCP), so we should use https://console.cloud.google.com/logs/viewer?project=sourcegraph-dev

File: handbook/engineering/cloud/index.md
Comment (nicksnyder) [reply handbook/engineering/cloud/index.md:33]: Yes, is this problem statement accurate?

Comment (nicksnyder) [reply handbook/engineering/cloud/index.md:33]: cc @keegancsmith @ryanslade @asdine

Comment (tsenart) [reply handbook/engineering/cloud/index.md:33]: LGTM

File: handbook/engineering/security/index.md
Comment (chayim) [reply handbook/engineering/security/index.md:20]: We wanted store security logs (to start) in Hosted ElasticSearch to get started. The other side of the same discussion is storing (also Elastic backed) in our own infrastructure, but creating a new GCP project separate from production and dev.

File: handbook/engineering/security/index.md
Comment (chayim) [inline handbook/engineering/security/index.md:18]: Vulnerabilities with a high severity, or a CVSS score of at least 4.0 are resolved.

File: handbook/engineering/security/index.md
Comment (chayim) [inline handbook/engineering/security/index.md:15]: All Docker images are continuously scanned for known security vulnerabilities and the security team is alerted as vulnerabilities are found.

File: handbook/engineering/security/index.md
Comment (chayim) [inline handbook/engineering/security/index.md:16]: All compute nodes are continuously scanned for known security vulnerabilities and the security team is alerted as vulnerabilities are found.

File: handbook/engineering/security/index.md
Comment (chayim) [inline handbook/engineering/security/index.md:23]: Normalize log format across services so that it is easier to correlate and search; this is a precursor to creating automated alerts from the logs.

File: handbook/engineering/security/index.md
Comment (chayim) [inline handbook/engineering/security/index.md:22]: Centralize workload and cloud (i.e. GCE) logs; this provides visibility from a systems and services perspective and can be used for investigating and correlating security events of interest.

**Review by chayim — CHANGES_REQUESTED** [review]
Added suggested text below.

File: handbook/engineering/security/index.md
Comment (tsenart) [reply handbook/engineering/security/index.md:20]: I think we need to align on one centralized logging solution for engineering at large (including security use-cases), so this needs to be discussed. Will there be an RFC?

Comment (keegancsmith) [reply handbook/engineering/security/index.md:20]: yeah please link me to this rfc. As a sidenote we have had pretty bad experiences with elastic search for logging in the past. Currently (and for quite a while) I imagine the amount of log data we generate can just be searched with grep and we can avoid solutions that require maintaining an index.

File: handbook/engineering/cloud/index.md
Comment (asdine) [reply handbook/engineering/cloud/index.md:33]: It looks good 👍🏼

File: handbook/engineering/security/index.md
Comment (ElizabethStirling) [inline handbook/engineering/security/index.md:34]: I would suspect 1-4 would be sufficient to unblock, but 5 & 6 seem closely coupled with 4, so it might be reasonable to do all of them together. I know @chayim wants to focus on logging relatively soon.

File: handbook/engineering/security/index.md
Comment (chayim) [reply handbook/engineering/security/index.md:20]: Can I ask what the issues were that were experienced in the past? Was it speed, of search, managing Elastic itself, of something else entirely?

Comment (keegancsmith) [reply handbook/engineering/security/index.md:20]: All 3. It was slow, it was tricky to update configuration for it to parse our structured logs and it would constantly die. All in all I don't understand using it for logs at our scale (or even much larger scales). If you are generating a few GB of logs a day, that is totally reasonable to just grep. We also won't be using extra features kibana provides like alerting, we should be sticking to prometheus for that.

Maybe someone with more experience it it could make it work well for us, but to be honest this seems like a problem that shouldn't require someone with specific experience :P

Comment (chayim) [reply handbook/engineering/security/index.md:20]: So we used it to build a large data pipeline for security logs, as did the folks over at [siemmonster](https://siemonster.com/community-edition/).  We had it up to about 100GB of logs a day across, with closing indices, and structuring. But, we were using logstash to prep the data in the first place.

I fully agree - Elastic experience makes a difference in making it behave rationally.  However, the security side does care about the alerting. Our goal/use case is to both have the centralized logs, and work down the path of correlation.

File: handbook/engineering/cloud/index.md
Comment (ryanslade) [reply handbook/engineering/cloud/index.md:33]: LGTM

File: handbook/engineering/security/index.md
Comment (nicksnyder) [reply handbook/engineering/security/index.md:18]:
```suggestion
1. Vulnerabilities with a high severity, or a CVSS score of at least 4.0 are resolved.
```

File: handbook/engineering/security/index.md
Comment (nicksnyder) [reply handbook/engineering/security/index.md:15]: (if you use the suggestion format then I can directly apply your suggestions)

```suggestion
1. All Docker images are continuously scanned for known security vulnerabilities and the security team is alerted as vulnerabilities are found.
```

File: handbook/engineering/security/index.md
Comment (nicksnyder) [reply handbook/engineering/security/index.md:23]:
```suggestion
1. Normalize log format across services so that it is easier to correlate and search; this is a precursor to creating automated alerts from the logs.
```

File: handbook/engineering/security/index.md
Comment (nicksnyder) [reply handbook/engineering/security/index.md:16]:
```suggestion
1. All compute nodes are continuously scanned for known security vulnerabilities and the security team is alerted as vulnerabilities are found.
```

File: handbook/engineering/security/index.md
Comment (nicksnyder) [inline handbook/engineering/security/index.md:17]:
```suggestion
```

File: handbook/engineering/security/index.md
Comment (nicksnyder) [inline handbook/engineering/security/index.md:35]:
```suggestion
1. We advertise a bounty for each unique vulnerability that allows an unauthorized person to gain access to our [test security repository](https://github.com/sourcegraph/security-test/blob/main/README.md) on Sourcegraph Cloud.
```

File: handbook/engineering/security/index.md
Comment (nicksnyder) [inline handbook/engineering/security/index.md:36]:
```suggestion
1. We run a time-bound capture the flag event where there are larger bounties for being able to gain access to our [test security repository](https://github.com/sourcegraph/security-test/blob/main/README.md) on Sourcegraph Cloud.
```

File: handbook/engineering/security/index.md
Comment (nicksnyder) [reply handbook/engineering/security/index.md:33]: I think the key point is that we find and resolve all the easy/obvious things before we start asking people to attack us and rewarding large bounties.

It seems like "adding centralized logging" won't actually yield new insights unless we are under active attack, so that is why I think it isn't a pre-requisite to starting to award bounties (the next goal). Do you agree or disagree?

File: handbook/engineering/security/index.md
Comment (nicksnyder) [reply handbook/engineering/security/index.md]: Implementation details should be discussed in a relevant RFC when we get to that milestone. For now, it is enough to say that we want to have a place where we can search over all of our logs.

File: handbook/engineering/security/index.md
Comment (nicksnyder) [reply handbook/engineering/security/index.md]: I updated the PR and added the reasons there. If you disagree, please comment on the new diff.

**Comment by nicksnyder** [comment]
I just pushed some updates based on a live discussion I had with Elizabeth and Chayim earlier this week.

I am going to error on the side of optimistically merging this instead of waiting for re-review since this is a non-trivial diff that touches two team pages and I think the content is a material improvement, even if it isn't perfect.

@chayim @ElizabethStirling If you see any improvements that could be made, please open a PR to propose them. Also, I would like you to come up with a forecast for when these things will get done: which of these have we started on? what is rough relative effort in engineer weeks?

**Pull request merged**
